### ISSUE TITLE
Bug - 2.4.0 source code not compilable with VS 2017

### ISSUE THREAD

**Issue by bovirus** [eibol/ffmpeg_batch#15]
@eibol

I tried to compile your source code with VS 2017.

I foddn a probable error in project configuration file

WindowsFormsApp1.sln

there is probably a misconfigure file path

Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WindowsFormsApp1", "**WindowsFormsApp1\WindowsFormsApp1.csproj**", "{01C90E3D-419D-4BF5-9FCC-FCEBE6B16840}"

for me should be

Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WindowsFormsApp1", "**WindowsFormsApp1.csproj**", "{01C90E3D-419D-4BF5-9FCC-FCEBE6B16840}"

With original string VS 2017 report an error that  WindowsFormsApp1\WindowsFormsApp1.csproj is not available.

With the changes the project is loaded right.

After that when I tried to compile it I got these errors

1>------ Build started: Project: WindowsFormsApp1, Configuration: Debug Any CPU ------
1>m:\Development\ffmpeg_batch-master\Form1.cs(2,29,2,35): error CS0234: The type or namespace name 'FileIO' does not exist in the namespace 'Microsoft.VisualBasic' (are you missing an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Form1.cs(22,18,22,20): error CS0234: The type or namespace name 'UI' does not exist in the namespace 'System.Web' (are you missing an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Program.cs(1,29,1,48): error CS0234: The type or namespace name 'ApplicationServices' does not exist in the namespace 'Microsoft.VisualBasic' (are you missing an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Program.cs(44,31,44,58): error CS0246: The type or namespace name 'WindowsFormsApplicationBase' could not be found (are you missing a using directive or an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Program.cs(53,13,53,41): error CS0246: The type or namespace name 'StartupNextInstanceEventArgs' could not be found (are you missing a using directive or an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Program.cs(52,33,52,54): error CS0115: 'SingleInstanceApp.OnStartupNextInstance(StartupNextInstanceEventArgs)': no suitable method found to override
1>m:\Development\ffmpeg_batch-master\Program.cs(72,33,72,49): error CS0115: 'SingleInstanceApp.OnCreateMainForm()': no suitable method found to override
1>m:\Development\ffmpeg_batch-master\Form1.cs(224,9,224,14): error CS0246: The type or namespace name 'Form3' could not be found (are you missing a using directive or an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Form1.cs(225,9,225,14): error CS0246: The type or namespace name 'Form4' could not be found (are you missing a using directive or an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Form1.cs(226,9,226,14): error CS0246: The type or namespace name 'Form5' could not be found (are you missing a using directive or an assembly reference?)
1>m:\Development\ffmpeg_batch-master\Form1.cs(227,9,227,15): error CS0246: The type or namespace name 'Form16' could not be found (are you missing a using directive or an assembly reference?)
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

Is your project compatible with VS2017?

Are there some special requirements to compile your project?

**Comment by eibol** [comment]
Hi, I'm using VS 2019, where it compiles fine, that may be the reason. 
Anyway, I will update the files with the latest version.

**Comment by bovirus** [comment]
@eilbol

Please reopen this iisue.

i confirmed both problem (tested with VS2019).

WindowsFormsApp1.sln file contain info about path\file name not right ("WindowsFormsApp1\WindowsFormsApp1.csproj).
One time fixed the problm about this file path, I can load in VS2019 the sln project without error.

But when I tried to compile with VS2019 I got the errors reported.

Please load the .sln project in VS2019 and try to compile.

Thanks.

**Comment by eibol** [comment]
I'll need to create a virtual machine with a fresh Windows to test it.

**Comment by bovirus** [comment]
Ok.

Did you use always Windows to compile it VS2019?

**Comment by eibol** [comment]
Yes, always Windows, though I made a mess once and had to edit the project files manually.
Anyway, you need to put all github files inside a subfolder named "FFBatch", and I've uploaded the solution file for testing.
https://github.com/eibol/ffmpeg_batch/blob/master/FFBatch_main1.sln

**Comment by bovirus** [comment]
@ibeol

Similar (but less problem)

- file FFBatch_main1.sln

in the line

Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "FFBatch_Converter", "**FFBatch\FFBatch_Converter.csproj**", "{01C90E3D-419D-4BF5-9FCC-FCEBE6B16840}"

the path FFBatch\FFBatch_Converter.csproj is wrong because in source tree **FFBatch_Converter.csproj** is not in FFBatch subfolder

the line should be

Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "FFBatch_Converter", "**FFBatch_Converter.csproj**", "{01C90E3D-419D-4BF5-9FCC-FCEBE6B16840}"

Then when I fix the issue I laod the project but I got tehse errros

Build started...
1>------ Build started: Project: FFBatch_Converter, Configuration: Debug Any CPU ------
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2202,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "AeroWizard, Version=2.2.9.0, Culture=neutral, PublicKeyToken=915e74f5d64b8f37, processorArchitecture=MSIL". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2202,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "ICSharpCode.SharpZipLib, Version=1.0.7.0, Culture=neutral, PublicKeyToken=1b03e6acf1164f73, processorArchitecture=MSIL". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2202,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Drawing.Common, Version=4.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(3325,5): error MSB3325: Cannot import the following key file: FFBatch_Converter_TemporaryKey.pfx. The key file may be password protected. To correct this, try to import the certificate again or manually install the certificate to the Strong Name CSP with the following key container name: VS_KEY_88EBFC24030D40B9
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(3325,5): error MSB3321: Importing key file "FFBatch_Converter_TemporaryKey.pfx" was canceled.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

If someone is wrong please explain with details how to arrange your source code.

**Comment by eibol** [comment]
Hi again, There are some issues here.
- The application uses some nuget packages, IICSharpCode.SharpZipLib, System.Drawing.Common and Aerowizard.
- You can finde these packages at nuget package manager inside Visual Studio.
- You need to create your own Code key for encryption, at menu Project -> Properties of FFBatch_Converter -> Signature.
- You may have to run Visual Studio 2019 with admin rights at least once, after key generation.

I hope that helps.

**Comment by bovirus** [comment]
@eibol 

Thanks.

I beleivw that you coukd be interesteing fro all users to create in README in github page a section

HOW TO COMPILE

and to add these info.

**Comment by eibol** [comment]
You're right, I'll include it at the end of the readme file.
